### ISSUE TITLE
Dot model generation not working when input_shape is not specified (Maybe just missing error message) 

### ISSUE THREAD

**Issue by dietriro** [tensorflow/tensorflow#40823]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04.4 / 20.04
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 2.2.0
- Python version: 3.8.2

**Current behavior**

When using an arbitrary network architecture but without **input_shape** being defined in the first layer, the `plot_model` and `model_to_dot` functions do not work properly (at least from my point of view). The dot model returned only contains a node for the model name (_sequential_ in this case), the saved image only contains this _sequential_-node as well. 

![image](https://user-images.githubusercontent.com/23249088/85840675-324ee080-b79d-11ea-9141-e3051175a657.png)

**Describe the expected behavior**

Using Python 3.7 with Keras (2.3.1) and Tensorflow (2.1.0) the functions would generate the model without specifying an input shape. With Tensorflow 2.2 this behavior broke.

Since the input_shape is an optional parameter this should not happen without throwing an error/exception in my opinion. If this is a desired behavior then I would at least expect an error message which would tell me that an input shape is missing. 

But if the model builds and trains without the input shape, then it should also be convertible to a dot format or image.

**Standalone code to reproduce the issue**

```
from keras.utils import plot_model
from keras.models import Sequential
from keras.layers import Conv2D

model = Sequential()

# Does not work
model.add(Conv2D(32, (3, 3), padding="same", activation="relu"))

# Works
# model.add(Conv2D(32, (3, 3), input_shape=(1, 100, 100), padding="same", activation="relu"))

plot_model(model, 'model.png')
```

**Output of model to dot**

Not working:

digraph G {
concentrate=True;
dpi=96;
rankdir=TB;
node [shape=record];
140054209647712 [label=sequential];
}

Working:

digraph G {
concentrate=True;
dpi=96;
rankdir=TB;
node [shape=record];
139879434642768 [label="conv2d_input: InputLayer"];
139879443480480 [label="conv2d: Conv2D"];
139879434642768 -> 139879443480480;
}

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with [TF v2.2](https://colab.research.google.com/gist/amahendrakar/839eabb0065cdcca00470c227089d2c3/40823-tf-keras.ipynb) and [TF-nightly](https://colab.research.google.com/gist/amahendrakar/02bc6476e501c577409ebfa638781fde/40823-tf-keras.ipynb#scrollTo=HH1dcQTf4HdH). However, output is different when using just [keras](https://colab.research.google.com/gist/amahendrakar/62fb86d0c32c556d6273e8631c5f0107/40823-keras.ipynb#scrollTo=HH1dcQTf4HdH) (not tf.keras). Please find the linked gist. 
 - keras output
![keras](https://user-images.githubusercontent.com/57165142/85890507-1b46d780-b80b-11ea-98ec-1f158f5df3f6.png)

- tf.keras output
![tf-keras](https://user-images.githubusercontent.com/57165142/85890519-23067c00-b80b-11ea-8cca-88756da26f1c.png)

Thanks!

**Comment by saikumarchalla** [comment]
Was able to reproduce  the issue using TF version 2.5. Please find the gist [here](https://colab.research.google.com/gist/saikumarchalla/af31bbdcb950ba3d615a8cc22de87d9f/untitled85.ipynb).Thanks!

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!
